apis/r/R/Init.R

```diff
@@ -17,6 +17,39 @@
                        sQuote(rpkg_lib_version), sQuote(soma_lib_version))
         stop(msg, call. = FALSE)
     }
+
+    # This is temporary for https://github.com/single-cell-data/TileDB-SOMA/issues/2407
+    # It will be removed once 2407 is complete.
+    if (Sys.getenv("SOMA_R_NEW_SHAPE") != "") {
+      .pkgenv[["use_current_domain_transitional_internal_only"]] <- TRUE
+      cdmsg <- " SOMA_R_NEW_SHAPE enabled"
+    } else {
+      .pkgenv[["use_current_domain_transitional_internal_only"]] <- FALSE
+      cdmsg <- ""
+    }
+     msg <- sprintf("TileDB Core version %s used by TileDB-R package, but TileDB-SOMA uses %s [%s]",
+                    sQuote(rpkg_lib_version), sQuote(soma_lib_version), sQuote(cdmsg))
```

Comment:
What is `msg` here for? It seems like it's being created and never used

Reply:
It was used in the startup message; I'll double-check I still am

Reply:
Fixed (on next push)